experimental/packages/opentelemetry-sdk-node/src/sdk.ts

```diff
@@ -121,11 +125,18 @@ export class NodeSDK {
     this._configuration = configuration;
 
     this._resource = configuration.resource ?? new Resource({});
-    this._resourceDetectors = configuration.resourceDetectors ?? [
-      envDetector,
-      processDetector,
-      hostDetector,
-    ];
+    let defaultDetectors: (Detector | DetectorSync)[] = [];
+    if (env.OTEL_NODE_RESOURCE_DETECTORS.length > 0) {
+      defaultDetectors = getResourceDetectorsFromEnv();
+    } else {
+      defaultDetectors = [envDetector, processDetector, hostDetector];
+      if (env.OTEL_NODE_EXPERIMENTAL_DEFAULT_SERVICE_INSTANCE_ID) {
+        defaultDetectors.push(serviceInstanceIDDetector);
+      }
```

Comment:
We can probably drop the concept of `OTEL_NODE_EXPERIMENTAL_DEFAULT_SERVICE_INSTANCE_ID`  and just use `OTEL_NODE_RESOURCE_DETECTORS`. 

If it is in the list, then that's the opt-in. If the list is empty or unset then we go with just `envDetector`, `processDetector` and `hostDetector` :slightly_smiling_face:

Reply:
Done